reasoner/planner/AnswerCountEstimator.java

```diff
@@ -392,13 +407,15 @@ private RelationModel(RelationConstraint relation, long relationTypeEstimate,
                 this.rolePlayerTypes = new HashMap<>();
 
                 relation.players().forEach(player -> {
```

Comment:
by the way, not sure if you need/want to use this, but the relation role player always should have a set of `inferredRoleTypes` even if there is not role type variable?

Reply:
Too complicated to be useful.

Reply:
Thank you!
Updated to consider.